### PR TITLE
phpunit test for wp_nonce_ays AND UPDATE CODE for wp_die_handler to p…

### PR DESCRIPTION
**Pull request by pbearne**
tests for wp_nonce_ays
AND UPDATE CODE for wp_die_handler to pass the response into WPDieException so that expectExceptionCode() call work

Trac ticket: https://core.trac.wordpress.org/ticket/53882

### CODE REVIEW

File: tests/phpunit/tests/functions/wp_nonce_ays.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		wp_nonce_ays( 'random_string' );
+
+	}
+
+	/**
+	 *
+	 */
+	public function test_wp_nonce_ays_log_out() {
+		$this->expectException( 'WPDieException' );
+		$this->expectExceptionMessageRegExp( '/You are attempting to log out of Test Blog<\/p><p>Do you really want to <a href="http:\/\/example\.org\/wp-login\.php\?action=logout&amp;_wpnonce=.{10}">log out<\/a>\?/m' );
```
Comment (jrfnl) [inline tests/phpunit/tests/functions/wp_nonce_ays.php:24]:
```suggestion
		$this->expectExceptionMessageMatches( '/You are attempting to log out of Test Blog<\/p><p>Do you really want to <a href="http:\/\/example\.org\/wp-login\.php\?action=logout&amp;_wpnonce=.{10}">log out<\/a>\?/m' );
```

Comment (jrfnl) [reply tests/phpunit/tests/functions/wp_nonce_ays.php:24]: The test suite has been upgraded for better PHPUnit cross-version compatibility. You can now use all PHPUnit 9.x assertion and expectation methods and the tests will still work on PHPUnit 5.x - 9.x.
For more details, see https://core.trac.wordpress.org/ticket/46149

Comment (pbearne) [reply tests/phpunit/tests/functions/wp_nonce_ays.php]: I saw that the tests were failing and fixed already

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in https://core.trac.wordpress.org/changeset/51608.

**Pull request closed without merging**